src/connectors/configure/__tests__/connectConfigure-test.ts

```diff
@@ -157,7 +166,7 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/configure/j
 
     refine({ hitsPerPage: 3, facets: ['rating'] });
 
-    expect(widget.getConfiguration(new SearchParameters({}))).toEqual(
+    expect(widget.getConfiguration!(new SearchParameters({}))).toEqual(
```

Comment:
To avoid the `!` inside the test we can override the definition to force the presence of `getConfiguration`. This is what we do inside the `index` widget to avoid the problem. The `index` widget doesn't use the factory though so it might be a bit more complicated to implement.

Reply:
Since we already use this pattern in other connectors tests, I think this can wait until we have a proper widget definition interface with only required lifecycle hooks. WDYT?

Reply:
Yep sure! It was not for this PR anyway?